r/R/dataset-factory.R

```diff
@@ -45,15 +45,18 @@ DatasetFactory$create <- function(x,
     return(dataset___UnionDatasetFactory__Make(x))
   }
 
-  path_and_fs <- get_path_and_filesystem(x, filesystem)
-  selector <- FileSelector$create(path_and_fs$path, allow_not_found = FALSE, recursive = TRUE)
-
   if (is.character(format)) {
     format <- FileFormat$create(match.arg(format), ...)
   } else {
     assert_is(format, "FileFormat")
   }
 
+  path_and_fs <- get_path_and_filesystem(x, filesystem)
+
+  if (length(path_and_fs$path) > 1) {
```

Comment:
We also want to support the length-1 (a single file) use case. So if we have a length-1 path, we need to check if it is a file or directory. At this point, `path_and_fs$fs` is guaranteed to be a FileSystem -> FileSystem has a GetFileInfo method -> FileInfo has a `type` active binding -> see enums.R for what FileTypes are.

Reply:
Done in 4929f99b5b96b8082f9f10c3eeaedc38f23f5287

Reply:
Done in 4929f99b5b96b8082f9f10c3eeaedc38f23f5287